File: Programmers/Level1/ReportResult/ReportResult.java
import java.util.*;

public class ReportResult {
	public int[] solution(String[] id_list, String[] report, int k) {
        int[] answer = new int[id_list.length];
        int[] num = new int[id_list.length];
        HashSet<String> rp = new HashSet<>(Arrays.asList(report));
        ArrayList<String> result = new ArrayList<>();
        
        for(String s : rp) {
        	num[Arrays.asList(id_list).indexOf(s.split(" ")[1])]++;
        }
        for(int i = 0; i < id_list.length; i++) {
        	if(num[i] >= k)
        		result.add(id_list[i]);
        }
        for(String s : rp) {
        	if (result.contains(s.split(" ")[1]))
        		answer[Arrays.asList(id_list).indexOf(s.split(" ")[0])]++;
        }
        return answer;
    }
	
	public static void main(String[] args) {
		ReportResult m = new ReportResult();
		String[] id_list = {"muzi", "frodo", "apeach", "neo"};
		String[] report = {"muzi frodo","apeach frodo","frodo neo","muzi neo","apeach muzi"};
		int[] s = m.solution(id_list,report,2);
        // System.out.println(s);  // 스태틱 메서드는 클래스를 이용하여 호출
    }
}

File: Programmers/Level1/ReportResult/reportResult.py
# def solution(id_list, report, k):
#     answer = [0] * len(id_list)
#     result = []
#     report = list(set(report))
#     cnt = [0] * len(id_list)
#
#     for i in report:
#         cnt[id_list.index(i.split()[1])] += 1
#     for i, c in enumerate(cnt):
#         if c >= k:
#             result.append(id_list[i])
#     for x in report:
#         if x.split()[1] in result:
#             answer[id_list.index(x.split()[0])] += 1
#     return answer

def solution(id_list, report, k):
    answer = [0] * len(id_list)
    result = {x:0 for x in id_list}
    report = [x.split() for x in list(set(report))]

    for x,y in report:
        result[y] += 1
    for x,y in report:
        if result[y] >= k:
            answer[id_list.index(x)] += 1
    return answer